src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImpl.java

```diff
@@ -1095,4 +1091,35 @@ private void setErrorData(SBApiResponse response, String errMsg) {
 		response.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
 	}
 
+	@Override
+	public Response getUserWFApplicationFieldsV2(String rootOrg, String org, String wid, SearchCriteria criteria) {
+		Response response = new Response();
+		response.put(Constants.STATUS, HttpStatus.OK);
+		response.put(Constants.MESSAGE, Constants.SUCCESSFUL);
+		List<Object[]> updatedFieldValues = wfStatusRepo.findWfFieldsForUserUpdated(criteria.getServiceName(), criteria.getApplicationStatus(), wid);
+		TypeReference<List<HashMap<String, Object>>> typeRef = new TypeReference<List<HashMap<String, Object>>>() {
+		};
+		List<Map<String, Object>> result = new ArrayList<>();
+		for (Object[] fields : updatedFieldValues) {
```

Comment:
If the user doesn't have any workflow request -- how the response is sent back in case of V2 ?
can you still add data with empty array ?

Reply:
This is done.